docs/user_guide/command_line_interface.rst

```diff
@@ -31,45 +31,48 @@ The available options:
 
 ::
 
- usage: preprocess [-h] [--files files [files ...]] [--dataset dataset] [--num_partitions num_partitions] [--overwrite]
- [--generate_config [generate_config]] [--format format] [--delim delim] [--dtype dtype] [--not_remap_ids]
- [--dataset_split dataset_split dataset_split] [--start_col start_col] [--num_line_skip num_line_skip]
- output_directory
-
- Preprocess Datasets
-
- positional arguments:
- output_directory      Directory to put graph data
-
- optional arguments:
- -h, --help            show this help message and exit
- --files files [files ...]
-     Files containing custom dataset
- --dataset dataset     Supported dataset to preprocess
- --num_partitions num_partitions
-     Number of partitions to split the edges into
- --overwrite           Overwrites the output_directory if this issetOtherwise, files with same the names will be treated as the data for current dataset.
- --generate_config [generate_config], -gc [generate_config]
-     Generates a single-GPU/multi-CPU/multi-GPU training configuration file by default.
-     Valid options (default to GPU): [GPU, CPU, multi-GPU]
- --format format       Format of data, eg. srd
- --delim delim, -d delim
-     Specifies the delimiter
- --dtype dtype         Indicates the numpy.dtype
- --not_remap_ids       If set, will not remap ids
- --dataset_split dataset_split dataset_split, -ds dataset_split dataset_split
-     Split dataset into specified fractions
- --start_col start_col, -sc start_col
-     Indicates the column index to start from
- --num_line_skip num_line_skip, -nls num_line_skip
-     Indicates number of lines to skip from the beginning
-
- Specify certain config (optional): [--<section>.<key>=<value>]
+    usage: preprocess [-h] [--download_directory download_directory] [--files files [files ...] | --dataset dataset] [--num_partitions num_partitions]
```

Comment:
isn't `preprocess` supposed to be used for general datasets? why did we call it `download_directory`; we should have self-explanatory flags like `input_data_directory` , `output_data_directory`; also what is `files`? these names need to change

Reply:
Yes ``preprocess`` can be used for general datasets.
This ``download_directory`` is only used for ``preprocess`` to store downloaded files for supported datasets. When preprocessing custom datasets, this directory is not used. I think it might be inaccurate to call such a directory as ``input_data_directory`` since users do not use it for inputing general dataset files.
When preprocessing general datasets, users are expected to input the path to all input files through the option ``files``. I should use a more self-explanatory flag for this option such as ``input_data_path``.

Reply:
We should split operations such as download etc from the `preprocess` util. The functionality of `preprocess` should be limited to reading an input graph of a well specified schema and returning the necessary output files that Marius requires to perform training. As simple as that. Please separate `download` to a separate utility in the Marius system.